packages/new-polymath-issuer/src/components/enhancers/ParseCsv.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { Component, ReactNode } from 'react';
+import { csvParser } from '@polymathnetwork/new-shared';
+
+export interface RenderProps {
+  data: csvParser.ResultProps;
+  setFile(file: File): void;
+  clearFile(): void;
+}
+
+export interface ParseResult {
+  data: csvParser.ResultProps;
+  config: Config;
+  errors: string[];
+  warnings: string[];
+}
+
+interface Config {
+  columns: Array<csvParser.Column>;
+  header?: boolean;
+  maxRows?: number;
+  strict?: boolean;
+  parseErrorMessage: string;
+  missingRequiredColumnsErrorMessage?: string;
+  extraColumnsErrorMessage?: string;
+  rowsExceedMaxLimitErrorMessage?: string;
+}
+
+interface Props {
+  config: Config;
+  render(output: RenderProps): ReactNode;
+}
+
+interface State {
+  data: csvParser.ResultProps;
+}
+
+export class ParseCsv extends Component<Props, State> {
+  public state = {
+    data: {
+      result: [],
+      totalRows: 0,
+      validRows: 0,
+      errorRows: 0,
+      ignoredRows: 0,
+      isFileValid: false,
+      errors: [],
+    },
+  };
+
+  public setFile = async (file: string | File) => {
+    try {
```

Comment:
Instead of wrapping the whole method in a try catch, I'd rather only wrap the part that is actually being handled, which is the line with the `await` call. This is so that we don't mix up unrelated errors in that handler